### ISSUE TITLE
Multiple ssh keys?

### ISSUE THREAD

**Issue by jkimbo** [getsentry/freight#43]
We want to use freight to deploy multiple repos and we'd also like to use [github deploy keys](https://developer.github.com/guides/managing-deploy-keys/#deploy-keys) to give freight access to the repositories since it seems like a perfect fit! However deploy keys can only be used to access a single repo so any thoughts about allowing freight to use multiple ssh keys?

**Comment by dcramer** [comment]
(I'm assuming this isnt supported in the code, but didnt check)

IMO it makes sense to allow a per-repo key to be stored.

That said, the simplest way to do this is just to create a dummy GitHub account and add the key there. It's far easier to manage than individual deploy keys and the only downside is if it gets compromised you need to rotate all keys. That said, the likelihood of it's key getting compromised and nothing else is so low that it's not a real concern (assuming you're not reusing the key).

**Comment by jkimbo** [comment]
Yep that is the solution we've gone for and you're right, practically it doesn't make much difference. This issue was just a query as to whether you've considered this and whether it would make sense. The code currently doesn't support this since you define an ssh key in the configuration and that is what gets used for each project.
